amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -167,6 +170,9 @@ public Scheduler(@NonNull final CheckpointConfig checkpointConfig,
         this.shardConsumerDispatchPollIntervalMillis = this.coordinatorConfig.shardConsumerDispatchPollIntervalMillis();
         this.parentShardPollIntervalMillis = this.coordinatorConfig.parentShardPollIntervalMillis();
         this.executorService = this.coordinatorConfig.coordinatorFactory().createExecutorService();
+        this.diagnosticEventHandler = new DefaultDiagnosticEventHandler();
```

Comment:
This could come from the config.

Reply:
Given that we don't want the customer to modify this handler, should this still go in the config?